modules/dnn/src/tensorflow/tf_importer.cpp

```diff
@@ -2092,72 +2092,105 @@ void TFImporter::populateNet(Net dstNet)
             {
-                LayerParams flattenLp;
-                std::string flattenName = name + "/flatten";
-                CV_Assert(layer_id.find(flattenName) == layer_id.end());
-                int flattenId = dstNet.addLayer(flattenName, "Flatten", flattenLp);
-                layer_id[flattenName] = flattenId;
-                connect(layer_id, dstNet, parsePin(layer.input(0)), flattenId, 0);
-
-                LayerParams reshapeLp;
-                std::string reshapeName = name + "/reshape";
-                CV_Assert(layer_id.find(reshapeName) == layer_id.end());
-                reshapeLp.set("axis", 0);
-                reshapeLp.set("num_axes", 1);
-                int newShape[] = {1, 1, -1};
-                reshapeLp.set("dim", DictValue::arrayInt(&newShape[0], 3));
-
-                int reshapeId = dstNet.addLayer(reshapeName, "Reshape", reshapeLp);
-                layer_id[reshapeName] = reshapeId;
-                connect(layer_id, dstNet, Pin(flattenName), reshapeId, 0);
-
-                LayerParams avgLp;
-                std::string avgName = name + "/avg";
-                CV_Assert(layer_id.find(avgName) == layer_id.end());
-                avgLp.set("pool", "ave");
-                // pooling kernel H x 1
-                avgLp.set("global_pooling_h", true);
-                avgLp.set("kernel_w", 1);
-                int avgId = dstNet.addLayer(avgName, "Pooling", avgLp);
-                layer_id[avgName] = avgId;
-                connect(layer_id, dstNet, Pin(reshapeName), avgId, 0);
-
-                LayerParams sliceLp;
-                std::string layerShapeName = name + "/slice";
-                CV_Assert(layer_id.find(layerShapeName) == layer_id.end());
-                sliceLp.set("axis", 0);
-                int begin[] = {0};
-                int size[] = {1};
-                sliceLp.set("begin", DictValue::arrayInt(&begin[0], 1));
-                sliceLp.set("size", DictValue::arrayInt(&size[0], 1));
-                int sliceId = dstNet.addLayer(layerShapeName, "Slice", sliceLp);
-                layer_id[layerShapeName] = sliceId;
-                connect(layer_id, dstNet, Pin(layer.input(0)), sliceId, 0);
-
-                if (!keepDims)
+                int axis = toNCHW(indices.at<int>(0));
+                if (axis == 0)
                 {
-                    LayerParams squeezeLp;
-                    std::string squeezeName = name + "/squeeze";
-                    CV_Assert(layer_id.find(squeezeName) == layer_id.end());
-                    squeezeLp.set("axis", 0);
-                    squeezeLp.set("end_axis", 1);
-                    int squeezeId = dstNet.addLayer(squeezeName, "Flatten", squeezeLp);
-                    layer_id[squeezeName] = squeezeId;
-                    connect(layer_id, dstNet, Pin(layerShapeName), squeezeId, 0);
-                    layerShapeName = squeezeName;
-                }
+                    LayerParams flattenLp;
```

Comment:
Is there a way to keep origin alignment?

Reply:
Just to minimize the changes

Reply:
Done